### PR TITLE
add modal for renaming note, and also add error handling

### PR DESCRIPTION
**Pull request by chuyouchia**
This PR adds the ability for users to:
1. Rename their notes
2. Error handling if the input is not supported
3. Prevent empty note names with error message
4. cleanup of some dead code

### CODE REVIEW

**Comment by samlhuillier** [comment]
If the current file is renamed while open, the old file name is recreated because of our write logic on file change. Same issue as #154. Maybe we could think about adding logic to change file path without a write old file path being triggered...

**Comment by samlhuillier** [comment]
Also, it looks like there could be some remaining work todo with directories: it let's you rename a directory but then errors when it tries to rename it

File: electron/main/Files/registerFilesHandler.ts [hunk 1]
```diff
@@ -109,6 +110,39 @@ export const registerFileHandlers = (
...
+
+  ipcMain.handle(
+    "rename-file-recursive",
+    async (event, renameFileProps: RenameFileProps) => {
+      const windowInfo = windowsManager.getWindowInfoForContents(event.sender);
+
+      if (!windowInfo) {
+        throw new Error("Window info not found.");
+      }
+      windowsManager.watcher?.unwatch(windowInfo?.vaultDirectoryForWindow);
```
[the remark this answers is not in the document]
Comment (chuyouchia) [reply electron/main/Files/registerFilesHandler.ts:125]: mostly a perf optimization when renaming large,  nested directories

**Review by samlhuillier — APPROVED** [review]
Looks good. Only slight concern is with re-indexing of a large directory rename, which could ultimately be done with rename operations in the database rather than having to embed each file again. Though this is an optimisation for the future imo.

**Comment by samlhuillier** [comment]
Screenshot 2024-03-30 at 13 52 49


Also small issue I'm seeing with windows when renaming directories which have children in them - I assume this is because Reor is locking  the child file/directories (with the watcher perhaps) so we can't rename the parent.

**Pull request merged**